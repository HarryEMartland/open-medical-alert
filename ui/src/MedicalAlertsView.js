import * as React from "react";
import AlertView from "./AlertView";

export default function MedicalAlertsView({data}) {
    return <div>
        {data.asthma && <AlertView message='I have asthma' link='https://www.nhs.uk/conditions/asthma/'/>}
        {data.diabetes1 && <AlertView message='I have diabetes (Type 1)' link='https://www.nhs.uk/conditions/type-1-diabetes/'/>}
        {data.diabetes2 && <AlertView message='I have diabetes (Type 2)' link='https://www.nhs.uk/conditions/type-2-diabetes/'/>}
        {data.epilepsy && <AlertView message='I have epilepsy' link='https://www.nhs.uk/conditions/epilepsy/'/>}
        {data.seizures && <AlertView message='I have seizures' link='https://www.nhs.uk/conditions/what-to-do-if-someone-has-a-seizure-fit/'/>}
        {data.hypertension && <AlertView message='I have hypertension' link='https://www.nhs.uk/conditions/high-blood-pressure-hypertension/'/>}
        {data.strokeHistory && <AlertView message='I have a stroke history' link='https://www.nhs.uk/conditions/stroke/'/>}
        {data.hearingLoss && <AlertView message='I have hearing loss' link='https://www.nhs.uk/conditions/hearing-loss/'/>}
        {data.alzheimers && <AlertView message="I have Alzheimer's" link='https://www.nhs.uk/conditions/alzheimers-disease/'/>}
    </div>
}