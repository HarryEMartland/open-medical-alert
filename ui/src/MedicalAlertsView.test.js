import React from 'react';
import {act} from "react-dom/test-utils";
import {render} from "react-dom";
import MedicalAlertsView from "./MedicalAlertsView";
import exampleData from '../../server/data.example.json'


describe('MedicalAlertsView', () => {
    let container = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    it('should render with empty data', () => {
        const data = {}
        act(() => {
            render(<MedicalAlertsView data={data}/>, container);
        })
        expect(container).toMatchSnapshot();

    })

    it('should render with data', () => {
        const data = {}
        act(() => {
            render(<MedicalAlertsView data={exampleData}/>, container);
        })
        expect(container).toMatchSnapshot();

    })
})