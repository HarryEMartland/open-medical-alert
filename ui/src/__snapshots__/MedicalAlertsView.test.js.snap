// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicalAlertsView should render with data 1`] = `
<div>
  <div>
    <div
      class="alert alert-danger"
    >
      I have diabetes (Type 2)
      <a
        class="close"
        href="https://www.nhs.uk/conditions/type-2-diabetes/"
        target="_blank"
      >
        <span
          aria-hidden="true"
        >
          ⓘ
        </span>
      </a>
    </div>
    <div
      class="alert alert-danger"
    >
      I have seizures
      <a
        class="close"
        href="https://www.nhs.uk/conditions/what-to-do-if-someone-has-a-seizure-fit/"
        target="_blank"
      >
        <span
          aria-hidden="true"
        >
          ⓘ
        </span>
      </a>
    </div>
    <div
      class="alert alert-danger"
    >
      I have hearing loss
      <a
        class="close"
        href="https://www.nhs.uk/conditions/hearing-loss/"
        target="_blank"
      >
        <span
          aria-hidden="true"
        >
          ⓘ
        </span>
      </a>
    </div>
    <div
      class="alert alert-danger"
    >
      I have Alzheimer's
      <a
        class="close"
        href="https://www.nhs.uk/conditions/alzheimers-disease/"
        target="_blank"
      >
        <span
          aria-hidden="true"
        >
          ⓘ
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`MedicalAlertsView should render with empty data 1`] = `
<div>
  <div />
</div>
`;
